speech/callback_helpers.go

```diff
@@ -362,3 +362,28 @@ func synthesizerFireEventBookmarkReached(handle C.SPXRECOHANDLE, eventHandle C.S
 	}
 	handler(*event)
 }
+
+var translationSynthesisCallbacks = make(map[C.SPXHANDLE]TranslationSynthesisEventHandler)
+
+func registerTranslationSynthesizingCallback(handler TranslationSynthesisEventHandler, handle C.SPXHANDLE) {
+	mu.Lock()
+	defer mu.Unlock()
+	translationSynthesisCallbacks[handle] = handler
+}
+
+func getTranslationSynthesizingCallback(handle C.SPXHANDLE) TranslationSynthesisEventHandler {
+	mu.Lock()
+	defer mu.Unlock()
+	return translationSynthesisCallbacks[handle]
+}
+
+//export translatorFireEventSynthesizing
+func translatorFireEventSynthesizing(handle C.SPXRECOHANDLE, eventHandle C.SPXEVENTHANDLE) {
+	handler := getTranslationSynthesizingCallback(handle)
+	event, err := NewTranslationSynthesisEventArgsFromHandle(handle2uintptr(eventHandle))
+	if err != nil || handler == nil {
+		C.recognizer_event_handle_release(handle)
```

Comment:
Why is only this branch releasing? Doesn't seem to get referenced anywhere in the function.

Reply:
`handle` is released if handler is nil or error, but doesn't seem to be referenced or released if they are not?

Reply:
i think eventHandle should be released here?

Reply:
Would make more sense. Would still need to be freed by the handler after handling it. Not sure who's supposed to be the owner - not finding any use of the function.

Reply:
event handle is released from consumer side